fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/CDU/A320_Neo_CDU_AirwaysFromWaypointPage.js

```diff
@@ -102,11 +98,17 @@ class A320_Neo_CDU_AirwaysFromWaypointPage {
                     rows[i] = [`${pendingAirway.ident}[color]cyan`, "[\xa0\xa0\xa0][color]cyan"];
 
                     mcdu.onRightInput[i] = (value, scratchpadCallback) => {
-                        const targetPlan = mcdu.flightPlan(forPlan, inAlternate);
+                        const currentPlan = mcdu.flightPlan(forPlan, inAlternate);
 
                         if (value.length > 0) {
                             Fmgc.WaypointEntryUtils.getOrCreateWaypoint(mcdu, value, false).then(/** @param wp {import('msfs-navdata').Fix | undefined} */ (wp) => {
                                 if (wp) {
+                                    if (!currentPlan.pendingAirways) {
+                                        mcdu.flightPlanService.startAirwayEntry(prevFpIndex, forPlan, inAlternate);
+                                    }
+
+                                    const targetPlan = mcdu.flightPlan(forPlan, inAlternate);
```

Comment:
How is `targetPlan` different from `currentPlan`? Will `targetPlan` be a temporary plan?

Reply:
They're the same thing; I originally was not declaring `currentPlan`, and only declaring `targetPlan`.

I ran into an issue where if I only assigned `const targetPlan = mcdu.flightPlan...` once, either at the top or after the `startAirwayEntry` then the respective results would be that either `targetPlan.pendingAirways.thenAirway` would be undefined or `if (!targetPlan.pendingAirways) {...}` would be undefined.

I'm pretty sure it's because `const targetPlan = mcdu.flightPlan...` needs to be assigned after `mcdu.flightPlanService.startAirwayEntry(prevFpIndex, forPlan, inAlternate);` or it won't pick up the new pending airway entry. At the same time though, if I don't assign `mcdu.flightPlan` before doing `if (!targetPlan.pendingAirways) {...}` then that if statement is undefined because `targetPlan` is shadowed under `mcdu.onLeftInput` and out of scope of `mcdu.onRightInput`.

Of course, I can't use the same variable name twice here if I'm reassigning, hence the two different names, and I'm admittedly not seeing a way of preserving the logic without redeclaring the variable.